pkg/plugin/connector/builtin/dispenser.go

```diff
@@ -49,12 +49,12 @@ func (d *Dispenser) DispenseSpecifier() (connector.SpecifierPlugin, error) {
 	return newSpecifierPluginAdapter(d.specifierPlugin(), d.pluginLogger("specifier")), nil
 }
 
-func (d *Dispenser) DispenseSource() (connector.SourcePlugin, error) {
-	return newSourcePluginAdapter(d.sourcePlugin(), d.pluginLogger("source")), nil
+func (d *Dispenser) DispenseSource(connectorID string) (connector.SourcePlugin, error) {
```

Comment:
It looks like a dispenser is always dispensing either sources or destinations, so I think I can get the connector ID into the dispenser as a field, and keep the method signature as it was (without any params).